report/reportserver/server.go

```diff
@@ -239,12 +229,18 @@ func (s *Server) HandleWorkspaceUpdate(event reportevents.ReportEvent) {
 
 	case *reportevents.WorkspaceError:
 		log.Println("[TRACE] Got workspace error event", *e)
-		payload := buildWorkspaceErrorPayload(e)
+		payload, err := buildWorkspaceErrorPayload(e)
+		if err != nil {
```

Comment:
can we put the panic into a defer block so we only call it once

Reply:
i.e. set an `err` variable and break instead of panicking - then check `err` in the defer